contributions/MPI_IS_gaussian_process/src/gaussian_process_guider.cpp

```diff
@@ -274,7 +274,19 @@ double GaussianProcessGuider::result(double input, double SNR, double time_step,
         {
             dithering_active_ = false;
         }
-        deduceResult(time_step); // just pretend we would do dark guiding...
+        try
+        {
+            deduceResult(time_step); // just pretend we would do dark guiding...
+        }
+        catch (const std::runtime_error err)

```

Comment:
we can use pass-by-reference semantics here rather than pass-by-value:
```suggestion
        catch (const std::runtime_error& err)
```